exporters/otlp/all/src/main/java/io/opentelemetry/exporter/otlp/logs/OtlpGrpcLogRecordExporterBuilder.java

```diff
@@ -101,6 +101,26 @@ public OtlpGrpcLogRecordExporterBuilder setTimeout(Duration timeout) {
     return this;
   }
 
+  /**
+   * Sets the maximum time to wait for new connections to be established. If unset, defaults to
+   * {@value GrpcExporterBuilder#DEFAULT_CONNECT_TIMEOUT_SECS}s.
+   */
+  public OtlpGrpcLogRecordExporterBuilder setConnectTimeout(long timeout, TimeUnit unit) {
+    requireNonNull(unit, "unit");
+    checkArgument(timeout >= 0, "timeout must be non-negative");
```

Comment:
> I figured we get around this by only accepting positive values

does this mean you want:

```suggestion
    checkArgument(timeout > 0, "timeout must be positive");
```

Reply:
Yeah, I like what @trask suggested.

Reply:
Hmm.. on closer inspection we have this same problem with all the export timeout settings as well:
- https://github.com/open-telemetry/opentelemetry-java/blob/main/exporters/otlp/all/src/main/java/io/opentelemetry/exporter/otlp/trace/OtlpGrpcSpanExporterBuilder.java#L85
- https://github.com/open-telemetry/opentelemetry-java/blob/main/exporters/otlp/all/src/main/java/io/opentelemetry/exporter/otlp/http/trace/OtlpHttpSpanExporterBuilder.java#L53

I'd like to change all of these settings to enforce `timeout > 0` instead of `timeout >= 0`, but want to do it in a separate PR because its a behavioral change. I've opened an issue to track this work: https://github.com/open-telemetry/opentelemetry-java/issues/6245

I'd like to merge this PR as is, and go back and fix #6245 before this is released.

Reply:
🙈 Oof, well, I guess it was worth pointing this out after all. I'm ok with the approach of doing it in a follow-up PR. 👍🏻